src/conn/conn_api.c

```diff
@@ -2939,16 +2930,19 @@ wiredtiger_open(const char *home, WT_EVENT_HANDLER *event_handler, const char *c
      * THE TURTLE FILE MUST BE THE LAST FILE CREATED WHEN INITIALIZING THE DATABASE HOME, IT'S WHAT
      * WE USE TO DECIDE IF WE'RE CREATING OR NOT.
      */
-    WT_ERR(__wt_turtle_init(session));
-    WT_ERR(__wt_config_gets(session, cfg, "verify_metadata", &cval));
-    verify_meta = cval.val;
+    WT_WITH_BUCKET_STORAGE(conn->bstorage, session, {
+        WT_ERR(__wt_turtle_init(session));
+        WT_ERR(__wt_config_gets(session, cfg, "verify_metadata", &cval));
+        verify_meta = cval.val;
 
-    /* Only verify the metadata file first. We will verify the history store table later.  */
-    if (verify_meta) {
-        wt_session = &session->iface;
-        ret = wt_session->verify(wt_session, WT_METAFILE_URI, NULL);
-        WT_ERR(ret);
-    }
+        /* Only verify the metadata file first. We will verify the history store table later.  */
+        if (verify_meta) {
+            wt_session = &session->iface;
+            ret = wt_session->verify(wt_session, WT_METAFILE_URI, NULL);
+            WT_ERR(ret);
+        }
+    });
```

Comment:
This is an unusual usage of a `WITH` macro. I'd personally prefer the lines move to a function and we do `WT_WITH_BUCKET_STORAGE(..., ret = new_function...)` if you can make that not too messy or too many args.